connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -1658,11 +1659,20 @@ private void backoff(long ms) {
         backoffRetries = BACKOFF_RETRIES;
     }
 
-    private void startAndStop(Collection<Callable<Void>> callables) {
+    // Visible for testing
+    void startAndStop(Collection<Callable<Void>> callables) {
         try {
             startAndStopExecutor.invokeAll(callables);
         } catch (InterruptedException e) {
             // ignore
+        } catch (RejectedExecutionException e) {
+            // Shutting down. Just log the exception
+            if (stopping.get()) {
+                log.debug("RejectedExecutionException thrown while herder is shutting down. This could be " +
+                        "because startAndStopExecutor is either already shutdown or is full.");
```

Comment:
The [Executors::newFixedThreadPool Javadocs](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newFixedThreadPool-int-java.util.concurrent.ThreadFactory-) state that the executor operates off of an "unbounded queue", so it's misleading to state that this error could arise because the executor is full.

It would also probably help to be clear to users that this is not a sign that something's wrong with the worker.

```suggestion
                log.debug("Ignoring RejectedExecutionException thrown while starting/stopping connectors/tasks en masse " +
                        "as the herder is already in the process of shutting down. This is not indicative of a problem and is normal behavior");
```

Reply:
That's a good point. I didn't factor in the capacity part. Thanks for pointing it out 👍